internal/tfengine/schema.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfengine
+
+// TODO(https://github.com/golang/go/issues/35950): Move this to its own file.
+const schema = `
+title: Terraform Engine Config Schema
+
+properties:
+  data:
+    description: |
+      Global set of key-value pairs to pass to all templates.
+      It will be merged with data set in the templates.
+    type: object
+
+  templates:
+    description: |
+      Templates the engine will parse and fill in with values from data.
+      Templates use the Go templating engine: https://golang.org/pkg/text/template/
+      Helper template funcs are also defined in ../template/funcmap.go.
+    type: array
+    items:
+      type: object
+      additionalProperties: false
+      properties:
+        name:
+          description: Name of the template. Can be referred to by 'output_ref'.
+          type: string
+
+        recipe_path:
+          description: Path to a recipe YAML config. Mutually exclusive with 'component_path'.
+          type: string
+
+        output_ref:
+          description: |
+            OutputRef allows this template write its contents to the same dir as another template.
+
+            Example:
+            If output_ref is set to 'foo.bar' then this will look for sibling template 'foo' (defined in this list) and a child template inside 'foo' called 'bar'.
```

Comment:
I'm still unclear on how `output_ref` functions. How does the child template 'bar' get used in this case?

Reply:
I added more explanation, let me know if this is clearer. There will also be schemas for each recipe that will list all the output_refs for each recipe.